### PR TITLE
Fixed `MixMaterial.isReadyForSubMesh` to remove WebGL warnings

### PR DESCRIPTION
**Pull request by dad72**
Hi,
I took over this PR which corrected the TerrainMaterial and had the same problem with MixMaterial.
So I copy the patches on MixMaterial hoping it's good for you @julien-moreau 

https://github.com/BabylonJS/Babylon.js/commit/1a29f504d33f5031b4ec3aae90fadfd67c814e91

### CODE REVIEW

**Comment by deltakosh** [comment]
There is a lint issue (some trailing whitespaces): https://travis-ci.com/BabylonJS/Babylon.js/jobs/176531346

**Comment by dad72** [comment]
Ok, I to correct the extra spaces and I add a test if the mixMap2 is null to not test diffuse textures 5,6,7 and 8.

File: materialsLibrary/src/mix/mixMaterial.ts [hunk 1]
```diff
@@ -171,20 +171,48 @@ export class MixMaterial extends PushMaterial {
 
         // Textures
         if (scene.texturesEnabled) {
+            if (this._mixTexture1 && !this._mixTexture1.isReady()) {
```
Comment (julien-moreau) [inline materialsLibrary/src/mix/mixMaterial.ts]: Would say:
if (!mixmap1 or !mixmap1.ready) return false

File: materialsLibrary/src/mix/mixMaterial.ts [hunk 2]
```diff
@@ -171,20 +171,48 @@ export class MixMaterial extends PushMaterial {
...
+                    return false;
+                }
+                if (!this._diffuseTexture3 || !this._diffuseTexture3.isReady()) {
+                    return false;
+                }
+                if (!this._diffuseTexture4 || !this._diffuseTexture4.isReady()) {
+                    return false;
+                }
+
+                if (this._mixTexture2 !== null) {
```
Comment (julien-moreau) [inline materialsLibrary/src/mix/mixMaterial.ts]: Would say:
if (this._mixTexture2) { ... }
In case of users assign « undefined » instead of null

**Comment by julien-moreau** [comment]
Added a last review, then everything looks good to me :)

File: materialsLibrary/src/mix/mixMaterial.ts (hunk 2 above)
Comment (dad72) [reply materialsLibrary/src/mix/mixMaterial.ts]: Yes you have reasons. I am going to correct.

**Comment by dad72** [comment]
Should be good now.

File: materialsLibrary/src/mix/mixMaterial.ts [hunk 3]
```diff
@@ -171,20 +171,48 @@ export class MixMaterial extends PushMaterial {
...
+            if (!this._mixTexture1 || !this._mixTexture1.isReady()) {
+                return false;
+            }
+
+            defines._needUVs = true;
+
+            if (this._mixTexture2) {
+                defines.MIXMAP2 = true;
+            }
+            if (!this._mixTexture2 || !this._mixTexture2.isReady()) {
```
Comment (julien-moreau) [inline materialsLibrary/src/mix/mixMaterial.ts]: My bad, I think this test should move in line 205
If (mixTexture2) { ... }
To test that if mixTexture2 then check if mixmap2 is ready and all other diffuse textures for mixmap2. You can also add the define in this test

**Comment by dad72** [comment]
Ok.

File: materialsLibrary/src/mix/mixMaterial.ts [hunk 4]
```diff
@@ -171,20 +171,47 @@ export class MixMaterial extends PushMaterial {
...
+                if (!this._diffuseTexture3 || !this._diffuseTexture3.isReady()) {
+                    return false;
+                }
+                if (!this._diffuseTexture4 || !this._diffuseTexture4.isReady()) {
+                    return false;
                 }
+
                 if (this._mixTexture2) {
-                    if (!this._mixTexture2.isReady()) {
+                    if (!this._mixTexture2 || !this._mixTexture2.isReady()) {
```
Comment (julien-moreau) [inline materialsLibrary/src/mix/mixMaterial.ts]: Just tested that mixTexture2 exists, so only isReady should be tested
Else, everything looks perfect for me :)

**Comment by dad72** [comment]
Like this ?

**Comment by julien-moreau** [comment]
Good to me !

**Comment by dad72** [comment]
No, I did not test it, but it should remove the WebGL error as you did on the TerrainEditor.

**Comment by julien-moreau** [comment]
Anyway looks ok
Can you fix the lint issue and update the what’s new? Thanks!

**Comment by dad72** [comment]
It is add.

**Comment by deltakosh** [comment]
we still have lint issues:
https://travis-ci.com/BabylonJS/Babylon.js/jobs/176649995

**Comment by dad72** [comment]
Decidedly ... I find it annoying all its spaces that adds itself Why can not it be removed automatically by compiling it?

Should be good now.

**Comment by julien-moreau** [comment]
All checks passed, you rox :)

**Comment by dad72** [comment]
Cool. Ready to be validated so. :+1:

**Comment by deltakosh** [comment]
Thanks a lot!!!
